src/Service/Controllers/RestController.cs

```diff
@@ -47,15 +49,17 @@ public class RestController : ControllerBase
         public const string REDIRECTED_ROUTE = "favicon.ico";
 
         private readonly ILogger<RestController> _logger;
+        private readonly TelemetryClient? _telemetryClient;
 
         /// <summary>
         /// Constructor.
         /// </summary>
-        public RestController(RestService restService, IOpenApiDocumentor openApiDocumentor, ILogger<RestController> logger)
+        public RestController(RestService restService, IOpenApiDocumentor openApiDocumentor, ILogger<RestController> logger, TelemetryClient? telemetryClient = null)
```

Comment:
if log events emmitted using ILogger are pushed to app insights, why do we also need `TelemetryClient`? Couldn't we just add additional `ILogger.Info` events with the desired custom properties?

Reply:
telemetry client is required for tracking event, ILogger would just send logs. For creating an event in ApplicationInsights, we need to use `TrackEvent` method provided by `Microsoft.ApplicationInsights` package.